python/taichi/lang/kernel_impl.py

```diff
@@ -641,10 +641,12 @@ def __call__(self, *args, **kwargs):
 
 
 def _inside_class(level_of_class_stackframe):
-    frames = oinspect.stack()
     try:
-        maybe_class_frame = frames[level_of_class_stackframe]
-        statement_list = maybe_class_frame[4]
+        maybe_class_frame = inspect.currentframe()
+        while level_of_class_stackframe > 0:
```

Comment:
I'd prefer a for-loop to be more pythonic:

```python
for _ in range(level_of_class_stackframe):
```

Reply:
Updated :-)